src/retasc/product_pages_api.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+"""
+Production Pages API
+"""
+
+from functools import cache
+
+from retasc import requests_session
+
+
+class ProductPagesApi:
+    """
+    Product Pages API Client
+    """
+
+    def __init__(self, api_url: str):
+        self.api_url = api_url
+        self.session = requests_session.requests_session()
+
+    def api_url_releases(self, release_name: str | None = None) -> str:
+        return f"{self.api_url.rstrip('/')}/releases/{release_name or ''}"
+
+    def api_url_release_schedule_tasks(self, release_name: str) -> str:
+        return f"{self.api_url_releases(release_name)}/schedule-tasks"
+
+    @cache
+    def active_releases(
+        self,
+        product_shortname: str,
+        fields: str = "shortname,product_shortname,ga_date,phase_display",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}?product__shortname={product_shortname}&phase__lt=Unsupported&fields=a,b,c
+        The response have the following fields:shortname,product_shortname,ga_date,phase_display
+        phase__lt=Unsupported means "any supported phase" (these are less than Unsupported).
+        """
+
+        opt = {
+            "product__shortname": product_shortname,
+            "phase__lt": "Unsupported",
+            "fields": fields,
+        }
+        res = self.session.get(self.api_url_releases(), params=opt)
+        res.raise_for_status()
+        return res.json()
+
+    @cache
+    def release_schedules(
+        self,
+        release_short_name: str,
+        fields: str = "name,date_start,date_finish,flags",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks?fields=a,b,c
+        The response have the following fields:name,date_start,date_finish,flags
+        """
+
+        url = self.api_url_release_schedule_tasks(release_short_name)
+        res = self.session.get(url, params={"fields": fields})
+        res.raise_for_status()
+        return res.json()
+
+    @cache
+    def release_schedules_by_search_query(
+        self, release_short_name: str, **kargs
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks/?{flag_cond}={flags}&{name__regex}={reg_query}&fields=a,b,c
+        """
+
+        url = self.api_url_release_schedule_tasks(release_short_name)
+        # fields has some default value, can be updated if kargs has it
+        addtional_query = {
+            "fields": "release_shortname,name,slug,date_start,date_finish"
+        }
+        for k, v in kargs.items():
+            addtional_query.update({k: v})
```

Comment:
```suggestion
    def release_schedules_by_search_query(
        self, release_short_name: str, fields: str = "release_shortname,name,slug,date_start,date_finish", **kwargs
    ) -> list[dict]:
        """
        https://{api_url_release_schedule_tasks}/?{kwargs...}
        """

        url = self.api_url_release_schedule_tasks(release_short_name)
```
 
But this looks the same as `release_schedules()` except `fields` is different and support for additional kwargs.

Can you add support for the kwargs to `release_schedules()` and remove `release_schedules_by_search_query()` method?